internal/controllers/cluster/cluster_controller_status_test.go

```diff
@@ -0,0 +1,2336 @@
+	cp := &unstructured.Unstructured{Object: map[string]interface{}{
+		"kind": "FakeControlPlane",
+		"metadata": map[string]interface{}{
+			"name": name,
+		},
+	}}
+
+	for _, opt := range options {
+		opt.ApplyToControlPlane(cp)
+	}
+	return cp
+}
+
+type fakeInfraClusterOption interface {
+	ApplyToInfraCluster(cp *unstructured.Unstructured)
+}
+
+func fakeInfraCluster(name string, options ...fakeInfraClusterOption) *unstructured.Unstructured {
+	cp := &unstructured.Unstructured{Object: map[string]interface{}{
+		"kind": "FakeInfraCluster",
+		"metadata": map[string]interface{}{
+			"name": name,
+		},
+	}}
+
+	for _, opt := range options {
+		opt.ApplyToInfraCluster(cp)
+	}
+	return cp
+}
+
+type fakeMachinePoolOption interface {
+	ApplyToMachinePool(mp *expv1.MachinePool)
+}
+
+func fakeMachinePool(name string, options ...fakeMachinePoolOption) *expv1.MachinePool {
+	mp := &expv1.MachinePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt.ApplyToMachinePool(mp)
+	}
+	return mp
+}
+
+type fakeMachineDeploymentOption interface {
+	ApplyToMachineDeployment(md *clusterv1.MachineDeployment)
+}
+
+func fakeMachineDeployment(name string, options ...fakeMachineDeploymentOption) *clusterv1.MachineDeployment {
+	md := &clusterv1.MachineDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt.ApplyToMachineDeployment(md)
+	}
+	return md
+}
+
+type fakeMachineSetOption interface {
+	ApplyToMachineSet(ms *clusterv1.MachineSet)
+}
+
+func fakeMachineSet(name string, options ...fakeMachineSetOption) *clusterv1.MachineSet {
+	ms := &clusterv1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt.ApplyToMachineSet(ms)
+	}
+	return ms
+}
+
+type fakeMachinesOption interface {
+	ApplyToMachine(ms *clusterv1.Machine)
+}
+
+func fakeMachine(name string, options ...fakeMachinesOption) *clusterv1.Machine {
+	m := &clusterv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	for _, opt := range options {
+		opt.ApplyToMachine(m)
+	}
+	return m
+}
+
+type controlPlaneRef corev1.ObjectReference
+
+func (r controlPlaneRef) ApplyToCluster(c *clusterv1.Cluster) {
+	c.Spec.ControlPlaneRef = ptr.To(corev1.ObjectReference(r))
+}
+
+type infrastructureRef corev1.ObjectReference
+
+func (r infrastructureRef) ApplyToCluster(c *clusterv1.Cluster) {
+	c.Spec.InfrastructureRef = ptr.To(corev1.ObjectReference(r))
+}
+
+type infrastructureReady bool
+
+func (r infrastructureReady) ApplyToCluster(c *clusterv1.Cluster) {
+	c.Status.InfrastructureReady = bool(r)
+}
+
+type controlPlaneReady bool
+
+func (r controlPlaneReady) ApplyToCluster(c *clusterv1.Cluster) {
+	c.Status.ControlPlaneReady = bool(r)
+}
+
+type desiredReplicas int32
+
+func (r desiredReplicas) ApplyToControlPlane(cp *unstructured.Unstructured) {
+	_ = unstructured.SetNestedField(cp.Object, int64(r), "spec", "replicas")
```

Comment:
Maybe use 	contract.ControlPlane().Replicas().Set()

(similar for other cases below)